src/js/modules/autocomplete.js

```diff
@@ -9,228 +9,464 @@ import memoize from './memoize.js';
  * Forked and modified from https://github.com/xavi/miss-plete
  */
 class Autocomplete {
- /**
- * class Constructor Autocomplete.
- */
-  constructor({
-    selector,
-    options,
-    classname,
-    scoreFn = memoize(Autocomplete.scoreFn),
-    listItemFn = Autocomplete.listItemFn
-  }) {
-    Object.assign(this, {selector, options, classname, scoreFn, listItemFn});
+  /**
+   * @param   {object} settings  Configuration options
+   * @return  {this}             The class
+   * @constructor
+   */
+  constructor(settings = {}) {
+    this.settings = {
+      'selector': settings.selector, // required
+      'options': settings.options, // required
+      'classname': settings.classname, // required
+      'selected': (settings.hasOwnProperty('selected')) ?
+        settings.selected : false,
+      'score': (settings.hasOwnProperty('score')) ?
+        settings.score : memoize(Autocomplete.score),
+      'listItem': (settings.hasOwnProperty('listItem')) ?
+        settings.listItem : Autocomplete.listItem,
+      'getSiblingIndex': (settings.hasOwnProperty('getSiblingIndex')) ?
+        settings.getSiblingIndex : Autocomplete.getSiblingIndex
+    };
+
     this.scoredOptions = null;
     this.container = null;
     this.ul = null;
-    this.highlightedIndex = -1;
+    this.highlighted = -1;
+
+    this.SELECTORS = Autocomplete.selectors;
+    this.STRINGS = Autocomplete.strings;
+    this.MAX_ITEMS = Autocomplete.maxItems;
+
+    window.addEventListener('keydown', (e) => {
+      this.keydownEvent(e);
+    });
+    window.addEventListener('keyup', (e) => {
+      this.keyupEvent(e);
+    });
+    window.addEventListener('input', (e) => {
+      this.inputEvent(e);
+    });
 
-    window.addEventListener('keyup', (event) => {
-      if (!event.target.matches(this.selector)) return;
-      this.input = event.target;
+    window.addEventListener('change', (e) => {
+      this.scrollUpToElement(e);
```

Comment:
Instead of adding this to the change event, can you add it to to the selected() method below. Otherwise it doesn't seem to fire until you blur from the element.